src/cryptonote_core/blockchain.cpp

```diff
@@ -4110,6 +4131,45 @@ bool get_network_block_database_hash(std::vector<std::string> &block_verifiers_d
     message_string = "{\r\n \"message_settings\": \"NODE_TO_BLOCK_VERIFIERS_GET_RESERVE_BYTES_DATABASE_HASH\",\r\n \"block_height\": \"" + std::to_string(current_block_height) + "\",\r\n}";
   }
 
+  std::string current_block_verifiers_list_public_address = string.substr(string.find("\"block_verifiers_public_address_list\": \"")+40,(string.find("\"",string.find("\"block_verifiers_public_address_list\": \"")+40)) - (string.find("\"block_verifiers_public_address_list\": \"")+40));
+  std::vector<std::string> ip_addresses=split(current_block_verifiers_list_IP_address.data(),"|");
+  std::vector<std::string> public_addresses=split(current_block_verifiers_list_public_address.data(),"|");
+  if(ip_addresses.size()!=public_addresses.size()){
+      MGINFO_RED( "IP address length need same as public_address length");//Better have a retry refresh a get list
+      return false;
+  }
+  std::string ip_address;
+  std::string public_address;
+  int passRootRequest = 0;
+  std::string result;
+  for(uint64_t i=0; i<public_addresses.size(); i++)
+  {
+     ip_address=ip_addresses[i];
+     public_address=public_addresses[i];
+     if(public_address==NETWORK_DATA_NODE_PUBLIC_ADDRESS_1||public_address==NETWORK_DATA_NODE_PUBLIC_ADDRESS_2||public_address==NETWORK_DATA_NODE_PUBLIC_ADDRESS_3||public_address==NETWORK_DATA_NODE_PUBLIC_ADDRESS_4||public_address==NETWORK_DATA_NODE_PUBLIC_ADDRESS_5){
+        result = send_and_receive_data(ip_address,message_string,SEND_OR_RECEIVE_SOCKET_DATA_DOWNLOAD_DATABASE_HASH_WEAK_TIMEOUT_SETTINGS);//Some not root node server my drop port
+        if (result != NODE_TO_BLOCK_VERIFIERS_GET_RESERVE_BYTES_DATABASE_HASH_ERROR_MESSAGE && result != "")
+        {
+          block_verifiers_database_hashes[passRootRequest] = result == NODE_TO_BLOCK_VERIFIERS_GET_RESERVE_BYTES_DATABASE_HASH_ERROR_MESSAGE || result == "" ? "" : result;
+          passRootRequest++;
+          if(passRootRequest>2){
+           break;
+          }
+        }
+     }
+  }
+  if(passRootRequest>=2){
+      hashesStatus=1;
+      MGINFO_RED("Pass root node validate");
+      return true;
+    }
+  if(passRootRequest==0){
+     MGINFO_RED("At least need include one root node sign it");
+     if(OPEN_SAFE_VALIDATE==1){
```

Comment:
I dont fully understand what OPEN_SAFE_VALIDATE does

It looks to me like your saying "check the response quickly on all nodes, if its a root node see if it responded. If 0 root nodes responded then cancel going any further with verifying this block"?

The seed nodes dont have to be online for the chain to continue so I dont understand this code either?

Reply:
If 27+ requests are executed, some requests actually will time out. If the root node is stable, the connection timeout(60s) caused by other node problems can be reduced. Sometimes we will download block data from some random nodes that are different from the root node. Some bad data lead to must download the blockchain reset data.If keep the data consistent with the root node can avoid this.